### ISSUE TITLE
Error while create VPN user for another account

### ISSUE THREAD

**Issue by edikevich** [apache/cloudstack#5711]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
UI?
~~~

##### CLOUDSTACK VERSION
~~~
4.16.0
~~~

##### CONFIGURATION
XCP-NG 8.2
Advanced Zone

##### OS / ENVIRONMENT
Oracle Linux 8.4 - Management server, MYSQL 8.0.26
XCP-NG 8.2 - hypervisor

##### SUMMARY
Error while create VPN user for another account:

![image](https://user-images.githubusercontent.com/42668237/143004325-5c0213a4-3f57-4bd3-b6db-a7498924cb2f.png)

![image](https://user-images.githubusercontent.com/42668237/143005490-84186e5c-2be8-4cad-936d-a4a94cdd4653.png)
 
but user was added:

![image](https://user-images.githubusercontent.com/42668237/143005870-26c92d47-a00d-46b1-a235-de0e4ff4cfa7.png)

and when U delete it U wil get error:

![image](https://user-images.githubusercontent.com/42668237/143006093-fdc72e1e-6157-47f6-8c86-419ad6e5e791.png)

##### STEPS TO REPRODUCE
~~~
Create new Isolated network. Enable VPN. Try to add new VPN user.
~~~
##### EXPECTED RESULTS
~~~
Can create VPN user. Can create VPN user for another account. 
~~~

##### ACTUAL RESULTS
~~~
Can create VPN user (if U dont specify domain and account). Can't create VPN user for another account (if U specify Doamin and Account). 
~~~
After we "creat" user in DB we can see:

![image](https://user-images.githubusercontent.com/42668237/143006549-a758bb5c-6f5b-4c73-ad5b-11b5bbd0ef74.png)

After we "remove" user:

![image](https://user-images.githubusercontent.com/42668237/143006752-77496b14-d225-4ff3-94dd-1ef073de7748.png)

Only delete from database allow create same named VPN user.

**Comment by shwstppr** [comment]
@edikevich Error on adding user seems to be okay when the owner does not have an associated VPN. However, it should not add the account altogether.
I've create a PR (#5850 ) which should fix this behaviour.

**Comment by sureshanaparti** [comment]
Fixed in #5850. @edikevich Closing this ticket as fix PR got merged. If you still notice this issue after the fix, please reopen. Thanks.
